.github/workflows/auto-update-dependencies.yml

```diff
@@ -0,0 +1,44 @@
+name: auto-update-dependencies
+
+on:
+  push:
+    branches:
+      - develop
+
+permissions:
+  contents: read
+
+env:
+  BRANCH_NAME: 'develop'
+
+jobs:
+  run:
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22
+      - name: Install dependencies
+        run: npm install axios --no-save
+      - name: Run dependency update script
+        run: node scripts/generateDependenciesList.js
+      - name: Do not commit yarn file
+        run: |
+          git checkout -- yarn.lock
+      - name: Auto-commit updated dependencies file
+        run: |
+          echo "Committing docs/DEPENDENCIES.md changes..."
+          FILE="docs/DEPENDENCIES.md"
+          CONTENT=$(base64 -w 0 "$FILE")
+          SHA=$(curl -s -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" "https://api.github.com/repos/${{ github.repository }}/contents/$FILE?ref=$BRANCH_NAME" | jq -r .sha)
+          if [ "$SHA" == "null" ]; then
+            SHA=""
+          fi
+          curl -s -X PUT -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
+            -d "{\"message\":\"Commit updated dependencies file [auto-generated]\", \"content\":\"$CONTENT\", \"branch\":\"$BRANCH_NAME\", \"sha\":\"$SHA\"}" \
```

Comment:
Note: if you look through commit history on this PR, you'll see something like this:
![image](https://github.com/user-attachments/assets/c29da103-ba1a-4bda-8604-f46e89b0dbba)

I ended up changing this message to `Commit updated dependencies file [auto-generated]` and that is what is going to be seen on the future PRs.